example/main.go

```diff
@@ -2,152 +2,262 @@ package main
 
 import (
 	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
 	"encoding/json"
 	"fmt"
-	"github.com/Not-Diamond/go-notdiamond/pkg/clients/azure"
-	"github.com/Not-Diamond/go-notdiamond/pkg/clients/openai"
-	"github.com/Not-Diamond/go-notdiamond/pkg/clients/vertex"
 	"io"
 	"log"
 	"net/http"
+	"os"
 	"time"
 
-	"github.com/Not-Diamond/go-notdiamond/pkg/http/response"
-	"github.com/Not-Diamond/go-notdiamond/pkg/transport"
+	"github.com/joho/godotenv"
+)
 
-	"example/config"
+// AWS Signature V4 constants
+const (
+	awsAlgorithm   = "AWS4-HMAC-SHA256"
+	awsRequestType = "aws4_request"
+	awsService     = "bedrock-runtime"
+	awsContentType = "application/json"
+	awsAccept      = "application/json"
+	awsBedrockHost = "bedrock-runtime.%s.amazonaws.com"
+	awsBedrockPath = "/model/%s/invoke"
 )
 
-func main() {
-	// Load configuration
-	cfg := config.LoadConfig()
+// BedrockConfig holds AWS Bedrock configuration
+type BedrockConfig struct {
+	AccessKeyID     string
+	SecretAccessKey string
+	Region          string
+}
+
+// ClaudeRequest represents the request format for Claude models
+type ClaudeRequest struct {
+	AnthropicVersion string                   `json:"anthropic_version"`
+	MaxTokens        int                      `json:"max_tokens"`
+	Messages         []map[string]interface{} `json:"messages"`
```

Comment:
Now that Go has generics, they added a keyword equivalent to `interface{}`: `any`.
```suggestion
	Messages         []map[string]any `json:"messages"`
```

Reply:
Looking further, why not make a type? Can even be anonymous, like:

```suggestion
	Messages []struct{
		Role    string `json:"role"`,
		Content string `json:"content"`,
	} `json:"messages"`
```

Note: I may see why later on when it's used, maybe some other code expects a plain map.

**ASIDE**
If you ARE going to be using a lot of untyped JS maps, I've also found myself making a simple pkg/json2 package so it can be used all around the project. I've used the pattern of making my own package named after a builtin but adding 2 when I want to extend something, like `time2` or `strings2`, not sure how much

```go
package json2

// M is a JSON map.
type M map[string]any
```
so you can do:
```go
body := json2.M {
    "model": "claude…",
    "messages: []json2.M {
        "role"   : "user",
        "content": "Hello, world!",
     }
}
```
You'll see many people have done the same: https://github.com/search?q=%22type+M+map%5Bstring%5Dinterface%7B%7D%22&type=code

It might feel silly to write a whole package for this, but packages are cheap, and you might think of other helpers in the future to put in there—better than a "utils" package. Also it's common to use one letter types in this kind of package, like I've seen it for databases like `mgo.M{ ... }` being an alias for a mongo map so you don't have to keep typing `map[string]interface{}` over and over again. Whatever YOU think is clearer. maybe you'd prefer `json2.Map` or `json2.Object` or skipping this package altogether.